azurelinuxagent/common/utils/restutil.py

```diff
@@ -205,7 +214,7 @@ def _http_request(method, host, rel_uri, port=None, data=None, secure=False,
 
     logger.verbose("HTTP connection [{0}] [{1}] [{2}] [{3}]",
                    method,
-                   url,
+                   cleanup_sas_tokens_from_urls(url),
```

Comment:
Is this the only place we log url?

Reply:
This is where we explicitly log. Added the cleanup action in a couple of more places where exceptions are raised.

Reply:
Added in https://github.com/Azure/WALinuxAgent/pull/1277/files#diff-aa566f84668a2b9beb5c97dbcefda221R329 and https://github.com/Azure/WALinuxAgent/pull/1277/files#diff-aa566f84668a2b9beb5c97dbcefda221R337 as well.

Reply:
Thanks